ckanext/security/logic/action.py

```diff
@@ -67,6 +69,10 @@ def user_update(up_func, context, data_dict):
     ckanext-security: reset throttling information for updated users
     to allow new login attempts after password reset
     """
+    # (canada fork only): update the user update form schema for username field
+    # TODO: upstream contrib??
+    if asbool(config.get('ckanext.security.use_ivalidators', False)):
+        context['schema']['name'].append(validators.old_username_validator)
```

Comment:
use `get_validator()` here so that other plugins can override the definition?

Reply:
put `old_username_validator` in plugin's `get_validators` and used toolkit here now.